## __Problema 3 (30 pts)__

La amitriptina es recetada por algunos médicos como un antidepresivo. Existen algunos efectos secundarios de esta droga como: pulsaciones cardiacas irregulares, presión arterial anormal, entre otras. Un conjunto de datos de 17 pacientes a los cuales se les administró esta droga fue recolectado en un estudio. 

Las variables respuestas fueron las siguientes:

$Y_1$ = Nivel de plasma Total

$Y_2$ = Cantidad de amitriptina presente en el plasma

Se midieron cinco variables prediuctoras

$Z_1$ = Género (1 Female, 0 Male) (GEN)

$Z_2$ = Cantidad de antidepresivo ingerido (AMT)

$Z_3$ = Medición de la onda cardiaca (PR)

$Z_4$ = Presión diastólica (DIAP)

$Z_5$ = Medida de onda cardiaca (QRS)

In [1]:
import numpy as np
import pandas as pd
import altair as alt
from sklearn.linear_model import LinearRegression

In [2]:
y_1 = np.array([3389, 1101, 1131, 596, 896, 1767, 807, 1111, 645, 628, 1360, 652, 860, 500, 781, 1070, 1754])
y_2 = np.array([3149, 653, 810, 448, 844, 1450, 493, 941, 547, 392, 1283, 458, 722, 384, 501, 405, 1520])

z_1 = np.array([1 ,1 ,0 ,1 ,1 ,1 ,1 ,0 ,1 ,1 ,1 ,1 ,1 ,0 ,0 ,0 ,1])
z_2 = np.array([7500, 1975, 3600, 675, 750, 2500, 350, 1500, 375, 1050, 3000, 450, 1750, 2000, 4500, 1500, 3000])
z_3 = np.array([220, 200, 205, 160, 185, 180, 154, 200, 137, 167, 180, 160, 135, 160, 180, 170, 180])
z_4 = np.array([0,  0,  60,  60,  70,  60,  80,  70,  60,  60,  60,  64,  90,  60,  0,  90,  0])
z_5 = np.array([140,  100,  111,  120,  83,  80,  98,  93,  105,  74,  80,  60,  79,  80,  100,  120,  129])

In [3]:
Z = np.array([z_1,z_2,z_3,z_4,z_5]).T
Y = np.array([y_1, y_2]).T

#### __a)__ Escriba un modelo de regresión con respuesta multivariada para abordar este problema.

In [4]:
reg = LinearRegression().fit(Z,Y)

In [5]:
reg.coef_

array([[6.75650781e+02, 2.84851063e-01, 1.02721328e+01, 7.25117136e+00,
        7.59823965e+00],
       [7.63029762e+02, 3.06373400e-01, 8.89619768e+00, 7.20555972e+00,
        4.98705075e+00]])

In [6]:
reg.intercept_

array([-2879.47824607, -2728.70854444])

#### __b)__ Determine los estimadores de m´ınimos cuadrados, valores ajustados y residuos usando los valores de la tabla con las observaciones. 

Luego $\pmb{\beta}$ es

In [7]:
beta=np.array([[-2879.47824607,6.75650781e+02, 2.84851063e-01, 1.02721328e+01, 7.25117136e+00,
        7.59823965e+00],[-2728.70854444,7.63029762e+02, 3.06373400e-01, 8.89619768e+00, 7.20555972e+00,
        4.98705075e+00]]).T

In [8]:
Z


array([[   1, 7500,  220,    0,  140],
       [   1, 1975,  200,    0,  100],
       [   0, 3600,  205,   60,  111],
       [   1,  675,  160,   60,  120],
       [   1,  750,  185,   70,   83],
       [   1, 2500,  180,   60,   80],
       [   1,  350,  154,   80,   98],
       [   0, 1500,  200,   70,   93],
       [   1,  375,  137,   60,  105],
       [   1, 1050,  167,   60,   74],
       [   1, 3000,  180,   60,   80],
       [   1,  450,  160,   64,   60],
       [   1, 1750,  135,   90,   79],
       [   0, 2000,  160,   60,   80],
       [   0, 4500,  180,    0,  100],
       [   0, 1500,  170,   90,  120],
       [   1, 3000,  180,    0,  129]])

In [9]:
beta

array([[-2.87947825e+03, -2.72870854e+03],
       [ 6.75650781e+02,  7.63029762e+02],
       [ 2.84851063e-01,  3.06373400e-01],
       [ 1.02721328e+01,  8.89619768e+00],
       [ 7.25117136e+00,  7.20555972e+00],
       [ 7.59823965e+00,  4.98705075e+00]])

$$\hat{\pmb{\beta}}= \begin{pmatrix} -2879.47825 & -2728.70854 \\ 
 675.650781 &  763.029762\\
 0.284851063 &  0.306373400\\
10.2721328 &  8.89619768 \\
7.25117136 &  7.20555972\\
7.59823965&  4.98705075\end{pmatrix}$$

In [10]:
z_0 = np.ones(17)
Z=np.array([z_0,z_1,z_2,z_3,z_4,z_5]).T

In [11]:
aux=Y-np.dot(Z,beta)
sigma = 1/17*np.dot(aux.T,aux)
sigma

array([[51176.95944034, 45039.79270646],
       [45039.79270646, 55335.81761062]])

$$\hat{\pmb{\Sigma}}=\begin{pmatrix}51176.95944034& 45039.79270646 \\45039.79270646& 55335.81761062 \end{pmatrix}$$

In [12]:
Yh=np.dot(Z,beta)
Yh.T

array([[3256.17827443, 1173.00390936, 1530.24768748,  978.84729006,
        1048.39128633, 1400.21355003,  802.50005268,  816.44319108,
         543.15932201,  808.05234438, 1542.63908153,  487.86610732,
         934.26617793,  376.69458153, 1011.15940643,  858.45510483,
        1479.88254278],
       [2987.47231216,  917.35329356, 1183.85243641,  695.29456456,
         828.21223101, 1232.86794316,  576.74210138,  478.27599171,
         323.96423667,  643.05363882, 1386.05464316,  355.95974344,
         813.93873841,  138.72752756,  749.99241296,  490.15162596,
        1198.08654671]])

$$\hat{\pmb{Y}}^T=\begin{pmatrix}3256.17827443& 1173.00390936& 1530.24768748&  978.84729006&
        1048.39128633& 1400.21355003&  802.50005268&  816.44319108&
         543.15932201&  808.05234438& 1542.63908153&  487.86610732&
         934.26617793&  376.69458153& 1011.15940643&  858.45510483&
        1479.88254278 \\ 2987.47231216&  917.35329356& 1183.85243641&  695.29456456&
         828.21223101& 1232.86794316&  576.74210138&  478.27599171&
         323.96423667&  643.05363882& 1386.05464316&  355.95974344&
         813.93873841&  138.72752756&  749.99241296&  490.15162596&
        1198.08654671\end{pmatrix}$$

In [13]:
epsh=Y-Yh
epsh.T

array([[ 132.82172557,  -72.00390936, -399.24768748, -382.84729006,
        -152.39128633,  366.78644997,    4.49994732,  294.55680892,
         101.84067799, -180.05234438, -182.63908153,  164.13389268,
         -74.26617793,  123.30541847, -230.15940643,  211.54489517,
         274.11745722],
       [ 161.52768784, -264.35329356, -373.85243641, -247.29456456,
          15.78776899,  217.13205684,  -83.74210138,  462.72400829,
         223.03576333, -251.05363882, -103.05464316,  102.04025656,
         -91.93873841,  245.27247244, -248.99241296,  -85.15162596,
         321.91345329]])

$$\hat{\pmb{\epsilon}}=\begin{pmatrix} 132.82172557,  161.52768784 \\ 
132.82172557,  161.52768784\\
-399.24768748, -373.85243641\\
-382.84729006, -247.29456456\\
-152.39128633,   15.78776899\\
366.78644997,  217.13205684\\
4.49994732,  -83.74210138\\
294.55680892,  462.72400829\\
101.84067799,  223.03576333\\
-180.05234438, -251.05363882\\
-182.63908153, -103.05464316\\
 164.13389268,  102.04025656\\
 -74.26617793,  -91.93873841\\
 123.30541847,  245.27247244\\
 -230.15940643, -248.99241296\\
211.54489517,  -85.15162596\\
274.11745722,  321.91345329\end{pmatrix}$$

#### __c)__ Realice un análisis de residuos

#### __d)__ Construya una región de confianza del 95% para ambas variables respuestas para $z_1=1,z_2=1200,z_3=1400,z_4=70,z_5=85$

Definiendo $z_0^T=(1,1,1200,1400,70,85)$

Se tiene que la región vendra dada por $$z_0^T\hat{\pmb{\beta_{(i)}}}\pm \sqrt{\left[\frac{m(n-r-1)}{n-r-m}F_{(m,n-r-m)}\right]}\sqrt{z_0^T(Z^TZ)^{-1}z_0\left(\frac{n}{n-r-1}\hat{\sigma}_{ii}\right)}$$

Notando que $n=17,r=5,m=2$

para $i=1$




In [14]:

aux2=np.linalg.inv(np.dot(Z.T,Z))
z=np.array([1,1,1200,1400,70,85])
np.dot(np.dot(z.T,aux2),z)

350.18992408496064

In [15]:
np.dot(z.T,beta.T[0].T)

13672.41209598

$$
\begin{aligned}
13672.41\pm \sqrt{\frac{2\cdot 11}{10}F_{(2,14)}}\sqrt{350.19\frac{17}{10}51176.96}
\end{aligned}$$

$$
\begin{aligned}
13672.41\pm \sqrt{\frac{2\cdot11}{10}3.739}\sqrt{350.19\frac{17}{10}51176.96}
\end{aligned}$$

In [16]:
np.dot(z.T,beta.T[1].T)

11784.93454371

para i=2
$$
\begin{aligned}
11784.93\pm \sqrt{\frac{2\cdot 11}{10}F_{(2,14)}}\sqrt{350.19\frac{17}{14}55335.81}
\end{aligned}$$

$$
\begin{aligned}
11784.93\pm \sqrt{\frac{2\cdot11}{10}3.739}\sqrt{350.19\frac{17}{10}55335.81}
\end{aligned}$$